INFO 2024-06-03 22:43:40,958 autoreload 19356 6268 Watching for file changes with StatReloader
INFO 2024-06-03 22:44:00,780 basehttp 19356 3540 "GET /about/ HTTP/1.1" 200 851
INFO 2024-06-03 22:44:03,121 basehttp 19356 3540 "GET /about/ HTTP/1.1" 200 851
INFO 2024-06-03 22:44:10,034 basehttp 19356 3540 "GET / HTTP/1.1" 200 1604
WARNING 2024-06-03 22:44:18,447 log 19356 3540 Not Found: /random_roll
WARNING 2024-06-03 22:44:18,448 basehttp 19356 3540 "GET /random_roll HTTP/1.1" 404 2439
INFO 2024-06-03 22:44:29,663 basehttp 19356 3540 "GET /about/ HTTP/1.1" 200 851
INFO 2024-06-11 21:14:12,786 autoreload 21064 2512 Watching for file changes with StatReloader
INFO 2024-06-11 21:14:17,786 basehttp 21064 19460 "GET / HTTP/1.1" 200 1604
INFO 2024-06-11 21:25:11,338 autoreload 14032 2724 Watching for file changes with StatReloader
INFO 2024-07-01 20:21:10,370 autoreload 5316 7312 Watching for file changes with StatReloader
INFO 2024-07-01 20:22:09,817 autoreload 20012 24796 Watching for file changes with StatReloader
INFO 2024-07-01 20:24:20,042 autoreload 20012 24796 D:\GB\Django\Djangofirstproject\homework\hw_2app\urls.py changed, reloading.
INFO 2024-07-01 20:24:20,465 autoreload 22000 6328 Watching for file changes with StatReloader
INFO 2024-07-01 20:58:35,021 autoreload 22000 6328 D:\GB\Django\Djangofirstproject\homework\hw_2app\urls.py changed, reloading.
INFO 2024-07-01 20:58:35,556 autoreload 8708 26484 Watching for file changes with StatReloader
INFO 2024-07-01 20:59:58,568 autoreload 8708 26484 D:\GB\Django\Djangofirstproject\homework\hw_2app\urls.py changed, reloading.
INFO 2024-07-01 20:59:58,996 autoreload 1608 23008 Watching for file changes with StatReloader
INFO 2024-07-01 21:00:24,875 autoreload 26616 5356 Watching for file changes with StatReloader
INFO 2024-07-01 21:00:27,031 basehttp 26616 2436 "GET / HTTP/1.1" 200 1604
WARNING 2024-07-01 21:00:27,167 log 26616 2436 Not Found: /favicon.ico
WARNING 2024-07-01 21:00:27,167 basehttp 26616 2436 "GET /favicon.ico HTTP/1.1" 404 2544
INFO 2024-07-01 21:00:49,866 autoreload 14768 22324 Watching for file changes with StatReloader
INFO 2024-07-01 21:00:51,596 basehttp 14768 19516 "GET / HTTP/1.1" 200 1604
WARNING 2024-07-01 21:00:51,652 log 14768 19516 Not Found: /favicon.ico
WARNING 2024-07-01 21:00:51,652 basehttp 14768 19516 "GET /favicon.ico HTTP/1.1" 404 2544
INFO 2024-07-01 21:01:13,972 basehttp 14768 19516 "GET / HTTP/1.1" 200 1604
WARNING 2024-07-01 21:01:21,462 log 14768 19516 Not Found: /shop
WARNING 2024-07-01 21:01:21,462 basehttp 14768 19516 "GET /shop HTTP/1.1" 404 2523
WARNING 2024-07-01 21:01:34,968 log 14768 19516 Not Found: /shop/orders
WARNING 2024-07-01 21:01:34,974 basehttp 14768 19516 "GET /shop/orders HTTP/1.1" 404 2626
WARNING 2024-07-01 21:01:46,334 log 14768 19516 Not Found: /shop/orders/2
WARNING 2024-07-01 21:01:46,334 basehttp 14768 19516 "GET /shop/orders/2 HTTP/1.1" 404 2632
WARNING 2024-07-01 21:03:46,123 log 14768 19516 Not Found: /shop/orders/2
WARNING 2024-07-01 21:03:46,125 basehttp 14768 19516 "GET /shop/orders/2 HTTP/1.1" 404 2632
WARNING 2024-07-01 21:03:49,188 log 14768 19516 Not Found: /shop/orders/2
WARNING 2024-07-01 21:03:49,188 basehttp 14768 19516 "GET /shop/orders/2 HTTP/1.1" 404 2632
WARNING 2024-07-01 21:03:52,617 log 14768 19516 Not Found: /shop/orders/2
WARNING 2024-07-01 21:03:52,617 basehttp 14768 19516 "GET /shop/orders/2 HTTP/1.1" 404 2632
INFO 2024-07-01 21:05:49,848 autoreload 10560 13676 Watching for file changes with StatReloader
INFO 2024-07-01 21:05:51,506 basehttp 10560 27624 "GET / HTTP/1.1" 200 1604
WARNING 2024-07-01 21:06:02,452 log 10560 27624 Not Found: /shop
WARNING 2024-07-01 21:06:02,458 basehttp 10560 27624 "GET /shop HTTP/1.1" 404 2523
WARNING 2024-07-01 21:07:53,615 log 10560 27624 Not Found: /shop
WARNING 2024-07-01 21:07:53,615 basehttp 10560 27624 "GET /shop HTTP/1.1" 404 2523
WARNING 2024-07-01 21:07:55,298 log 10560 27624 Not Found: /shop
WARNING 2024-07-01 21:07:55,298 basehttp 10560 27624 "GET /shop HTTP/1.1" 404 2523
WARNING 2024-07-01 21:07:56,177 log 10560 27624 Not Found: /shop
WARNING 2024-07-01 21:07:56,177 basehttp 10560 27624 "GET /shop HTTP/1.1" 404 2523
WARNING 2024-07-01 21:07:56,543 log 10560 27624 Not Found: /shop
WARNING 2024-07-01 21:07:56,543 basehttp 10560 27624 "GET /shop HTTP/1.1" 404 2523
WARNING 2024-07-01 21:07:56,762 log 10560 27624 Not Found: /shop
WARNING 2024-07-01 21:07:56,762 basehttp 10560 27624 "GET /shop HTTP/1.1" 404 2523
WARNING 2024-07-01 21:07:56,936 log 10560 27624 Not Found: /shop
WARNING 2024-07-01 21:07:56,936 basehttp 10560 27624 "GET /shop HTTP/1.1" 404 2523
WARNING 2024-07-01 21:08:08,907 log 10560 27624 Not Found: /shop/
WARNING 2024-07-01 21:08:08,916 basehttp 10560 27624 "GET /shop/ HTTP/1.1" 404 2608
WARNING 2024-07-01 21:12:12,331 log 10560 27624 Not Found: /post
WARNING 2024-07-01 21:12:12,331 basehttp 10560 27624 "GET /post HTTP/1.1" 404 2523
WARNING 2024-07-01 21:12:31,199 log 10560 27624 Not Found: /app3
WARNING 2024-07-01 21:12:31,199 basehttp 10560 27624 "GET /app3 HTTP/1.1" 404 2523
WARNING 2024-07-01 21:13:08,074 log 10560 27624 Not Found: /shop/orders
WARNING 2024-07-01 21:13:08,082 basehttp 10560 27624 "GET /shop/orders HTTP/1.1" 404 2626
WARNING 2024-07-01 21:13:19,876 log 10560 27624 Not Found: /shop/orders
WARNING 2024-07-01 21:13:19,879 basehttp 10560 27624 "GET /shop/orders HTTP/1.1" 404 2626
WARNING 2024-07-01 21:13:25,511 log 10560 27624 Not Found: /shop/orders/2
WARNING 2024-07-01 21:13:25,515 basehttp 10560 27624 "GET /shop/orders/2 HTTP/1.1" 404 2632
WARNING 2024-07-01 21:14:37,186 log 10560 27624 Not Found: /shop/ orders/2
WARNING 2024-07-01 21:14:37,186 basehttp 10560 27624 "GET /shop/%20orders/2 HTTP/1.1" 404 2637
INFO 2024-07-01 21:20:51,230 autoreload 19672 16700 Watching for file changes with StatReloader
INFO 2024-07-01 21:20:53,872 basehttp 19672 768 "GET / HTTP/1.1" 200 1604
INFO 2024-07-01 21:22:12,710 autoreload 19672 16700 D:\GB\Django\Djangofirstproject\homework\homework\urls.py changed, reloading.
INFO 2024-07-01 21:22:13,166 autoreload 21624 8272 Watching for file changes with StatReloader
WARNING 2024-07-01 21:22:15,905 log 21624 12096 Not Found: /
WARNING 2024-07-01 21:22:15,905 basehttp 21624 12096 "GET / HTTP/1.1" 404 2166
WARNING 2024-07-01 21:22:24,926 log 21624 12096 Not Found: /shop
WARNING 2024-07-01 21:22:24,926 basehttp 21624 12096 "GET /shop HTTP/1.1" 404 2196
WARNING 2024-07-01 21:22:35,011 log 21624 12096 Not Found: /shop/
WARNING 2024-07-01 21:22:35,019 basehttp 21624 12096 "GET /shop/ HTTP/1.1" 404 2281
WARNING 2024-07-01 21:22:47,854 log 21624 12096 Not Found: /shop/orders/2
WARNING 2024-07-01 21:22:47,856 basehttp 21624 12096 "GET /shop/orders/2 HTTP/1.1" 404 2305
INFO 2024-07-01 21:23:35,902 autoreload 21624 8272 D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py changed, reloading.
INFO 2024-07-01 21:23:36,358 autoreload 22020 24412 Watching for file changes with StatReloader
WARNING 2024-07-01 21:23:41,530 log 22020 940 Not Found: /
WARNING 2024-07-01 21:23:41,530 basehttp 22020 940 "GET / HTTP/1.1" 404 2166
WARNING 2024-07-01 21:23:53,955 log 22020 940 Not Found: /shop
WARNING 2024-07-01 21:23:53,955 basehttp 22020 940 "GET /shop HTTP/1.1" 404 2196
WARNING 2024-07-01 21:23:56,646 log 22020 940 Not Found: /shop
WARNING 2024-07-01 21:23:56,647 basehttp 22020 940 "GET /shop HTTP/1.1" 404 2196
WARNING 2024-07-01 21:24:05,803 log 22020 940 Not Found: /shop/orders
WARNING 2024-07-01 21:24:05,803 basehttp 22020 940 "GET /shop/orders HTTP/1.1" 404 2299
WARNING 2024-07-01 21:24:10,082 log 22020 940 Not Found: /shop/orders
WARNING 2024-07-01 21:24:10,083 basehttp 22020 940 "GET /shop/orders HTTP/1.1" 404 2299
WARNING 2024-07-01 21:24:11,848 log 22020 940 Not Found: /shop/orders
WARNING 2024-07-01 21:24:11,848 basehttp 22020 940 "GET /shop/orders HTTP/1.1" 404 2299
WARNING 2024-07-01 21:24:13,482 log 22020 940 Not Found: /shop/orders
WARNING 2024-07-01 21:24:13,488 basehttp 22020 940 "GET /shop/orders HTTP/1.1" 404 2299
INFO 2024-07-01 21:24:35,377 autoreload 22020 24412 D:\GB\Django\Djangofirstproject\homework\hw_2app\urls.py changed, reloading.
INFO 2024-07-01 21:24:35,835 autoreload 21848 27560 Watching for file changes with StatReloader
WARNING 2024-07-01 21:24:37,994 log 21848 25700 Not Found: /
WARNING 2024-07-01 21:24:37,994 basehttp 21848 25700 "GET / HTTP/1.1" 404 2166
WARNING 2024-07-01 21:24:46,144 log 21848 25700 Not Found: /shop
WARNING 2024-07-01 21:24:46,144 basehttp 21848 25700 "GET /shop HTTP/1.1" 404 2196
WARNING 2024-07-01 21:24:49,221 log 21848 25700 Not Found: /shop/
WARNING 2024-07-01 21:24:49,223 basehttp 21848 25700 "GET /shop/ HTTP/1.1" 404 2268
INFO 2024-07-01 21:24:55,560 basehttp 21848 25700 "GET /shop/orders HTTP/1.1" 301 0
INFO 2024-07-01 21:24:55,574 basehttp 21848 25892 "GET /shop/orders/ HTTP/1.1" 200 502
INFO 2024-07-01 21:31:16,023 autoreload 21848 27560 D:\GB\Django\Djangofirstproject\homework\hw_2app\urls.py changed, reloading.
INFO 2024-07-01 21:31:16,491 autoreload 13348 25988 Watching for file changes with StatReloader
WARNING 2024-07-01 21:31:35,840 log 13348 7208 Not Found: /
WARNING 2024-07-01 21:31:35,840 basehttp 13348 7208 "GET / HTTP/1.1" 404 2166
INFO 2024-07-01 21:31:47,925 basehttp 13348 7208 "GET /shop/orders/ HTTP/1.1" 200 431
INFO 2024-07-01 21:32:32,863 basehttp 13348 7208 "GET /shop/orders/ HTTP/1.1" 200 431
INFO 2024-07-01 21:32:33,983 basehttp 13348 7208 "GET /shop/ HTTP/1.1" 200 463
INFO 2024-07-01 21:35:58,687 autoreload 13348 25988 D:\GB\Django\Djangofirstproject\homework\hw_2app\urls.py changed, reloading.
INFO 2024-07-01 21:35:59,117 autoreload 3500 16004 Watching for file changes with StatReloader
INFO 2024-07-01 21:36:06,477 basehttp 3500 26260 "GET /shop/ HTTP/1.1" 200 463
WARNING 2024-07-01 21:36:12,187 log 3500 26260 Not Found: /shop/orders/
WARNING 2024-07-01 21:36:12,187 basehttp 3500 26260 "GET /shop/orders/ HTTP/1.1" 404 2475
ERROR 2024-07-01 21:36:15,051 log 3500 26260 Internal Server Error: /shop/orders/2
Traceback (most recent call last):
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py", line 23, in get_orders
    return render(request, 'hw_2app/orders.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\defaulttags.py", line 197, in render
    values = list(values)
             ^^^^^^^^^^^^
TypeError: 'ManyRelatedManager' object is not iterable
ERROR 2024-07-01 21:36:15,060 basehttp 3500 26260 "GET /shop/orders/2 HTTP/1.1" 500 161614
ERROR 2024-07-01 21:36:15,910 log 3500 20568 Internal Server Error: /shop/orders/2
Traceback (most recent call last):
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py", line 23, in get_orders
    return render(request, 'hw_2app/orders.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\defaulttags.py", line 197, in render
    values = list(values)
             ^^^^^^^^^^^^
TypeError: 'ManyRelatedManager' object is not iterable
ERROR 2024-07-01 21:36:15,918 basehttp 3500 20568 "GET /shop/orders/2 HTTP/1.1" 500 161340
ERROR 2024-07-01 21:39:10,949 log 3500 20568 Internal Server Error: /shop/orders/2
Traceback (most recent call last):
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py", line 23, in get_orders
    return render(request, 'hw_2app/orders.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\defaulttags.py", line 197, in render
    values = list(values)
             ^^^^^^^^^^^^
TypeError: 'ManyRelatedManager' object is not iterable
ERROR 2024-07-01 21:39:10,967 basehttp 3500 20568 "GET /shop/orders/2 HTTP/1.1" 500 161751
INFO 2024-07-01 21:39:49,951 basehttp 3500 20568 "GET /shop/orders/2 HTTP/1.1" 200 554
INFO 2024-07-01 21:42:57,395 basehttp 3500 20568 "GET /shop/orders/1 HTTP/1.1" 200 659
INFO 2024-07-01 21:49:48,087 autoreload 3500 16004 D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py changed, reloading.
INFO 2024-07-01 21:49:48,506 autoreload 21660 26716 Watching for file changes with StatReloader
ERROR 2024-07-01 21:49:51,845 log 21660 16864 Internal Server Error: /shop/orders/1
Traceback (most recent call last):
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py", line 18, in get_orders
    products = Product.objects.get()
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\db\models\query.py", line 652, in get
    raise self.model.MultipleObjectsReturned(
hw_2app.models.Product.MultipleObjectsReturned: get() returned more than one Product -- it returned 6!
ERROR 2024-07-01 21:49:51,851 basehttp 21660 16864 "GET /shop/orders/1 HTTP/1.1" 500 76499
ERROR 2024-07-01 21:50:47,540 log 21660 16864 Internal Server Error: /shop/orders/1
Traceback (most recent call last):
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py", line 18, in get_orders
    products = Product.objects.get()
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\db\models\query.py", line 652, in get
    raise self.model.MultipleObjectsReturned(
hw_2app.models.Product.MultipleObjectsReturned: get() returned more than one Product -- it returned 6!
ERROR 2024-07-01 21:50:47,540 basehttp 21660 16864 "GET /shop/orders/1 HTTP/1.1" 500 76499
ERROR 2024-07-01 21:50:49,431 log 21660 16864 Internal Server Error: /shop/orders/1
Traceback (most recent call last):
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py", line 18, in get_orders
    products = Product.objects.get()
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\db\models\query.py", line 652, in get
    raise self.model.MultipleObjectsReturned(
hw_2app.models.Product.MultipleObjectsReturned: get() returned more than one Product -- it returned 6!
ERROR 2024-07-01 21:50:49,440 basehttp 21660 16864 "GET /shop/orders/1 HTTP/1.1" 500 76499
INFO 2024-07-01 21:51:08,198 autoreload 21660 26716 D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py changed, reloading.
INFO 2024-07-01 21:51:08,748 autoreload 9808 24068 Watching for file changes with StatReloader
INFO 2024-07-01 21:51:13,681 basehttp 9808 5076 "GET /shop/orders/1 HTTP/1.1" 200 1095
INFO 2024-07-01 22:11:49,292 autoreload 9808 24068 D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py changed, reloading.
INFO 2024-07-01 22:11:49,693 autoreload 10620 22596 Watching for file changes with StatReloader
INFO 2024-07-01 22:21:19,517 autoreload 10620 22596 D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py changed, reloading.
INFO 2024-07-01 22:21:20,047 autoreload 22604 26328 Watching for file changes with StatReloader
INFO 2024-07-01 22:25:59,914 autoreload 22604 26328 D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py changed, reloading.
INFO 2024-07-01 22:26:00,392 autoreload 22116 1992 Watching for file changes with StatReloader
INFO 2024-07-01 22:30:51,392 autoreload 22116 1992 D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py changed, reloading.
INFO 2024-07-01 22:30:51,936 autoreload 2368 26136 Watching for file changes with StatReloader
INFO 2024-07-01 22:31:00,779 basehttp 2368 27436 "GET /shop/orders/1 HTTP/1.1" 200 692
ERROR 2024-07-01 22:37:08,080 log 2368 2356 Internal Server Error: /shop/orders/1
Traceback (most recent call last):
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 502, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'endblock'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\homework\hw_2app\views.py", line 28, in get_orders
    return render(request, 'hw_2app/orders.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader_tags.py", line 295, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 504, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "D:\GB\Django\Djangofirstproject\venv\Lib\site-packages\django\template\base.py", line 555, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 15: 'endblock', expected 'empty' or 'endfor'. Did you forget to register or load this tag?
ERROR 2024-07-01 22:37:08,080 basehttp 2368 2356 "GET /shop/orders/1 HTTP/1.1" 500 183464
INFO 2024-07-01 22:37:29,625 basehttp 2368 2356 "GET /shop/orders/1 HTTP/1.1" 200 634
INFO 2024-07-01 22:38:34,467 basehttp 2368 2356 "GET /shop/orders/1 HTTP/1.1" 200 634
INFO 2024-07-01 22:39:40,803 basehttp 2368 2356 "GET /shop/orders/1 HTTP/1.1" 200 673
INFO 2024-07-01 22:41:27,839 basehttp 2368 2356 "GET /shop/orders/1 HTTP/1.1" 200 708
INFO 2024-07-01 22:44:24,338 basehttp 2368 2356 "GET /shop/orders/1 HTTP/1.1" 200 820
INFO 2024-07-01 23:17:18,865 autoreload 15060 2636 Watching for file changes with StatReloader
INFO 2024-07-01 23:17:22,773 basehttp 15060 9128 "GET /shop/orders/1 HTTP/1.1" 200 1100
INFO 2024-07-01 23:22:59,056 autoreload 12824 21424 Watching for file changes with StatReloader
INFO 2024-07-01 23:23:02,235 basehttp 12824 21984 "GET /shop/orders/1 HTTP/1.1" 200 994
